guides/source/active_storage_overview.md

```diff
@@ -0,0 +1,305 @@
+* How to link to the attached file.
+* How to create variations of an image.
+* How to generate a preview for files other than images.
+* How to upload files directly to a service.
+* How to implement a download link.
+* How to add support for additional cloud services.
+* How to clean up files stored during testing.
+
+--------------------------------------------------------------------------------
+
+Active Storage makes it simple to upload and reference files in cloud services
+like Amazon S3, Google Cloud Storage, or Microsoft Azure Storage, and attach
+those files to Active Records. Supports having one main service and mirrors in
+other services for redundancy. It also provides a disk service for testing or
+local deployments, but the focus is on cloud storage.
+
+Files can be uploaded from the server to the cloud or directly from the client
+to the cloud.
+
+Image files can furthermore be transformed using on-demand variants for quality,
+aspect ratio, size, or any other
+[MiniMagick](https://github.com/minimagick/minimagick) supported transformation.
+
+## Compared to other storage solutions
+
+A key difference to how Active Storage works compared to other attachment
+solutions in Rails is through the use of built-in
+[Blob](https://github.com/rails/rails/blob/master/activestorage/app/models/active_storage/blob.rb)
+and
+[Attachment](https://github.com/rails/rails/blob/master/activestorage/app/models/active_storage/attachment.rb)
+models (backed by Active Record). This means existing application models do not
+need to be modified with additional columns to associate with files. Active
+Storage uses polymorphic associations via the `Attachment` join model, which
+then connects to the actual `Blob`.
+
+`Blob` models store attachment metadata (filename, content-type, etc.), and
+their identifier key in the storage service. Blob models do not store the actual
+binary data. They are intended to be immutable in spirit. One file, one blob.
+You can associate the same blob with multiple application models as well. And if
+you want to do transformations of a given `Blob`, the idea is that you'll simply
+create a new one, rather than attempt to mutate the existing one (though of
+course you can delete the previous version later if you don't need it).
+
+
+## Setup
+
+To setup an existing application after upgrading to Rails 5.2, run `rails active_storage:install`. If you're creating a new project with Rails 5.2, ActiveStorage will be installed by default. This generates the tables for the
+`Attachment` and `Blob` models.
+
+Inside a Rails application, you can set-up your services through the
+generated `config/storage.yml` file and reference one
+of the aforementioned constant under the +service+ key. For example:
+
+``` yaml
+  local:
+    service: Disk
+    root: <%= Rails.root.join("storage") %>
+```
+NOTE: Should we include the required keys for all the supported services?
+NOTE: Should we mention the mirror service and how to set it up?
```

Comment:
Yes and yes.

Reply:
To expand on my answer to the first question, let’s show the minimum required config for each service and refer readers to client library docs for optional config.